mybatis/runtime/src/main/java/io/quarkiverse/mybatis/runtime/QuarkusDatabaseIdProvider.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkiverse.mybatis.runtime;
+
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+import org.apache.ibatis.mapping.DatabaseIdProvider;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+public class QuarkusDatabaseIdProvider implements DatabaseIdProvider {
+    @Override
+    public String getDatabaseId(final DataSource dataSource) throws SQLException {
+        if (dataSource instanceof QuarkusDataSource ds) {
+            String name = ds.getDataSourceName();
+            return ConfigProvider.getConfig().getOptionalValue(
+                    "quarkus.mybatis." + ("<default>".equals(name) ? "" : name) + "database-id", String.class).orElse(null);
```

Comment:
I think it needs to be `"<default>".equals(name) ? "" : name+"."` to correctly handle named data sources - with current code if your datasource is named `x` this would read `quarkus.mybatis.xdatabase-id` instead of `quarkus.mybatis.x.database-id`

Reply:
good catch !